.github/workflows/build.yml

```
@@ -109,6 +109,83 @@ jobs:
+            echo "Regeneration touched .github/** — refusing to commit/push a workflow-poisoning diff."
+            git diff --name-only -- .github/
+            exit 1
+          fi
+
+      # Deliberately does NOT re-apply Renovate's `rebase` label after pushing. Re-adding it would
+      # tell Renovate to force-rebase the branch from the dependency update on its next run, which
+      # recreates the branch and discards this foreign commit — undoing the fix and looping forever.
+      # Renovate treats a foreign commit as a signal to stop auto-rebasing that branch, which is the
+      # outcome we want: our fix sticks, and the PR is left for CI to go green and get merged.
+      - name: Commit and push fixes
+        run: |
```

actionlint (1.7.12)

[error] "github.event.pull_request.head.ref" is potentially untrusted. avoid using it directly in inline scripts. instead, pass it through an environment variable. see https://docs.github.com/en/actions/reference/security/secure-use#good-practices-for-mitigating-script-injection-attacks for more details

(expression)